pkg/api/apis/operators/v1alpha1/installplan_types.go

```diff
@@ -145,6 +145,43 @@ type Step struct {
 	Status    StepStatus   `json:"status"`
 }
 
+// ManifestsMatch returns true if the CSV manifests in the StepResources of the given list of steps
+// matches those in the InstallPlanStatus.
+func (s *InstallPlanStatus) CSVManifestsMatch(steps []*Step) bool {
```

Comment:
if you make this a method on `installplan`, can't we skip half of this calculation? (`spec.clusterServiceVersionNames` is a list of the csvs)

Reply:
discussed in person - nevermind  😄